legal-api/src/legal_api/services/authz.py

```diff
@@ -537,6 +555,65 @@ def get_allowable_actions(jwt: JwtManager, business: Business):
     return result
 
 
+def get_could_file(business: Business,
+                   state: Business.State,
+                   legal_type: str,
+                   jwt: JwtManager):
+    """Get allowed type of filing types for the current user."""
+    # importing here to avoid circular dependencies
+    # pylint: disable=import-outside-toplevel
+    from legal_api.core.meta import FilingMeta
+
+    user_role = 'general'
+    if jwt.contains_role([STAFF_ROLE, SYSTEM_ROLE, COLIN_SVC_ROLE]):
+        user_role = 'staff'
+
+    allowable_filings = get_allowable_filings_dict().get(user_role, {}).get(state, {})
+    could_filing_types = []
+
+    for allowable_filing_key, allowable_filing_value in allowable_filings.items():
+        # skip if business does not exist and filing is not required
+        # skip if this filing does not need to be returned for existing businesses
+
+        business_status = allowable_filing_value.get('businessRequirement', BusinessRequirement.EXIST)
+
+        if business_status != BusinessRequirement.NO_RESTRICTION and \
+                bool(business) ^ (business_status == BusinessRequirement.EXIST):
+            continue
+
+        allowable_filing_legal_types = allowable_filing_value.get('legalTypes', [])
+
+        if allowable_filing_legal_types:
+            is_allowable = legal_type in allowable_filing_legal_types
+            allowable_filing_type = {'name': allowable_filing_key,
+                                     'displayName': FilingMeta.get_display_name(legal_type, allowable_filing_key),
+                                     'feeCode': Filing.get_fee_code(legal_type, allowable_filing_key)}
+            could_filing_types = add_allowable_filing_type(is_allowable,
+                                                           could_filing_types,
+                                                           allowable_filing_type)
+            continue
+
+        filing_sub_type_items = \
+            filter(lambda x: isinstance(x[1], dict) and legal_type in
+                   x[1].get('legalTypes', []), allowable_filing_value.items())
+
+        for filing_sub_type_item_key, _ in filing_sub_type_items:
+
+            allowable_filing_sub_type = {'name': allowable_filing_key,
+                                         'type': filing_sub_type_item_key,
+                                         'displayName': FilingMeta.get_display_name(legal_type,
+                                                                                    allowable_filing_key,
+                                                                                    filing_sub_type_item_key),
+                                         'feeCode': Filing.get_fee_code(legal_type,
```

Comment:
If you are planning to use `feeCode` values, there is one exception case (a dynamic fee code based on the user action in alteration. if alteration filing is to convert `BC to ULC` then fee code will be `NOALU` (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/models/filing.py#L116) and this fee code will not be available in get_fee_code function

Reply:
We aren't intending to use feeCode, I just wanted to keep the structure the same, and while I would have normally added another flag to allowedActions to remove the blockers check I figured this would be an easier review because it can't possibly affect the current use cases as they all use allowedActions